packages/test/test-end-to-end-tests/src/test/summarizeRestart.spec.ts

```diff
@@ -0,0 +1,203 @@
+
+describeNoCompat("Summarizer closes instead of refreshing", (getTestObjectProvider) => {
+	const settings = {};
+	const testContainerConfig: ITestContainerConfig = {
+		runtimeOptions: {
+			summaryOptions: {
+				summaryConfigOverrides: { state: "disabled" },
+			},
+		},
+		loaderProps: { configProvider: mockConfigProvider(settings) },
+	};
+
+	let provider: ITestObjectProvider;
+	const createContainer = async (): Promise<IContainer> => {
+		return provider.makeTestContainer(testContainerConfig);
+	};
+
+	beforeEach(async () => {
+		provider = getTestObjectProvider({ syncSummarizer: true });
+		settings["Fluid.ContainerRuntime.Test.SummarizationRecoveryMethod"] = "restart";
+	});
+
+	itExpects(
+		"Closes the summarizing client instead of refreshing",
+		[
+			{
+				eventName: "fluid:telemetry:Container:ContainerClose",
+				error: "Restarting summarizer instead of refreshing",
+			},
+		],
+		async () => {
+			const container = await createContainer();
+			const { container: summarizingContainer, summarizer } = await createSummarizer(
+				provider,
+				container,
+				undefined,
+				undefined,
+				mockConfigProvider(settings),
+			);
+
+			const dataStore = await requestFluidObject<ITestDataObject>(container, "default");
+			dataStore._root.set("an", "op");
+
+			await provider.ensureSynchronized();
+			const summarizeResults = summarizer.summarizeOnDemand({
+				reason: "end-to-end test",
+				refreshLatestAck: true,
+			});
+			await provider.ensureSynchronized();
+			await summarizeResults.receivedSummaryAckOrNack;
+			assert(summarizingContainer.closed, "Unknown acks should close the summarizer");
+			assert(!container.closed, "Original container should not be closed");
+		},
+	);
+
+	itExpects(
+		"Closes the summarizing client instead of refreshing with two clients",
+		[
+			{
+				eventName: "fluid:telemetry:Container:ContainerClose",
+				error: "Restarting summarizer instead of refreshing",
+			},
+			{
+				eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_cancel",
+				error: "Restarting summarizer instead of refreshing",
+			},
+			{
+				eventName: "fluid:telemetry:Summarizer:Running:HandleLastSummaryAckError",
+				error: "Restarting summarizer instead of refreshing",
+			},
+		],
+		async () => {
+			const container = await createContainer();
+			const { container: summarizingContainer, summarizer } = await createSummarizer(
+				provider,
+				container,
+				undefined,
+				undefined,
+				mockConfigProvider(settings),
+			);
+
+			const { container: summarizingContainer2, summarizer: summarizer2 } =
+				await createSummarizer(
+					provider,
+					container,
+					undefined,
+					undefined,
+					mockConfigProvider(settings),
+				);
+
+			await summarizeNow(summarizer);
+
+			const dataStore = await requestFluidObject<ITestDataObject>(container, "default");
+			dataStore._root.set("an", "op");
+
+			await provider.ensureSynchronized();
+			const container2 = summarizingContainer2 as any;
```

Comment:
You need to do all this hack, this test can be simple.
All you need is this:
- Create 2 summarizers.
- Let summarizer1 summarize and wait for the ack.
- Valiadate that summarizer2 closes on receiving the ack from summarizer1.

Reply:
I think I can just call summarizer2.run()